### ISSUE TITLE
Specify sidecar / asset format for text annotations and / or temporality

### ISSUE THREAD

**Issue by batpad** [sparkgeo/stac-video#23]
It would be nice to be able to associate textual annotations or temporal information about individual frames / timecode ranges in the video.

**Use Case**: I have a drone video that moves around a certain area and want to add arbitrary text annotations to particular time-code ranges to highlight certain aspects at particular times, etc.

**Use Case**: I have satellite imagery over a period of years composited into a video. I want to be able to indicate which frames / timecode ranges in the video correspond to what date of satellite imagery.

**Use Case**: For each frame of the video, I have some discrete data, and want to be able to associate some arbitrary data points for each frame and / or time-code range.

The current specification allows for changing geometries by specifying a geometry per frame as an array. For data that changes per frame, we could do something similar, and just allow more arbitrary data to be specified per-frame.

I do think it's useful to lean on established formats for associating time-based metadata to video for cases where the data should be associated with a "time-code range" and it doesn't really make sense to have it "per frame".

I think my proposal here is to specify something like `video:annotation:<annotation_type>` in assets to point to a file containing time-based annotations for the video.

There are a few different standards for representing time-based annotations to video:

 - `SRT`: This is the most popular format used, generally to provide subtitles for videos. It's an extremely simple text format, and has the advantage of a lot of existing tooling that works with it. It should be relatively straightforward to author using various existing tools.
 - `WebVTT`: Builds upon SRT, and is now a web standard: https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API - has native browser support and is very similar to SRT.

I think we should probably encourage WebVTT files for additional annotation tracks that users may want to associate with video.

@darrenwiens does this make sense? I can work on a PR if this seems sensible.

**Comment by darrenwiens** [comment]
Yes, that makes sense and it would amazing if you could put this together!
